rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -3201,9 +3202,11 @@ private Federation getFlyoverFederationFromTxOutputs(Keccak256 flyoverDerivation
             return activeFederation;
         }
 
-        Federation retiringFederation = getRetiringFederation();
-        if (retiringFederation != null && outputsMatchFlyoverFederation(btcTx, retiringFederation, flyoverDerivationHash)) {
-            return retiringFederation;
+        Optional<Federation> retiringFederation = getRetiringFederation()
+            .filter(retiringFed -> outputsMatchFlyoverFederation(btcTx, retiringFed, flyoverDerivationHash));
+
+        if (retiringFederation.isPresent()) {
+            return retiringFederation.get();
         }
 
         throw new IllegalStateException("Couldn't extract federation from btcTx outputs.");
```

Comment:
```suggestion
            return retiringFederation.get().orElseThrow(new IllegalStateException("Couldn't extract federation from btcTx outputs."););
}
```
Just makes it shorter